### PR TITLE
.Net: Simplify and clarify example 16

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Example is somewhat convoluted.

### Description
Remove unneeded code and comment remaining code.

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs [hunk 1]
```diff
@@ -3,130 +3,109 @@
...
 
 /**
- * The following example shows how to plug into SK a custom text generation model.
+ * The following example shows how to plug a custom text generation model into SK.
+ *
+ * To do this, this example uses a text generation service stub (MyTextGenerationService) and
+ * no actual model.
  *
- * This might be useful in a few scenarios, for example:
+ * Using a a custom text generation model within SK can be useful in a few scenarios, for example:
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:18]:
```suggestion
 * Using a custom text generation model within SK can be useful in a few scenarios, for example:
```

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs [hunk 2]
```diff
@@ -3,130 +3,109 @@
...
  * - You are not using OpenAI or Azure OpenAI models
- * - You are using OpenAI/Azure OpenAI models but the models are behind a web service with a different API schema
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:23]: nit: I think it's still could be valid scenario, we can rephrase it if needed.

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs [hunk 3]
```diff
@@ -3,130 +3,109 @@
...
         Kernel kernel = builder.Build();
 
-        const string FunctionDefinition = "Does the text contain grammar errors (Y/N)? Text: {{$input}}";
+        const string FunctionDefinition = "Write one paragraph on {{$input}}";
+        var ParagraphWritingFunction = kernel.CreateFunctionFromPrompt(FunctionDefinition);
 
-        var textValidationFunction = kernel.CreateFunctionFromPrompt(FunctionDefinition);
+        const string Input = "Why AI is awesome";
+        Console.WriteLine($"Function input: {Input}\n");
+        var result = await ParagraphWritingFunction.InvokeAsync(kernel, new());
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:52]: Looks like `Input` is not used for function execution:
```suggestion
        var result = await ParagraphWritingFunction.InvokeAsync(kernel, new(Input));
```

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs [hunk 4]
```diff
@@ -3,130 +3,109 @@
...
-        builder.Services.AddSingleton(ConsoleLogger.LoggerFactory);
         // Add your text generation service as a singleton instance
         builder.Services.AddKeyedSingleton<ITextGenerationService>("myService1", new MyTextGenerationService());
         // Add your text generation service as a factory method
         builder.Services.AddKeyedSingleton<ITextGenerationService>("myService2", (_, _) => new MyTextGenerationService());
         Kernel kernel = builder.Build();
 
-        const string FunctionDefinition = "Does the text contain grammar errors (Y/N)? Text: {{$input}}";
+        const string FunctionDefinition = "Write one paragraph on {{$input}}";
+        var ParagraphWritingFunction = kernel.CreateFunctionFromPrompt(FunctionDefinition);
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:48]: nit:
```suggestion
        var paragraphWritingFunction = kernel.CreateFunctionFromPrompt(FunctionDefinition);
```

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 3 above)
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:52]: nit:
```suggestion
        var result = await paragraphWritingFunction.InvokeAsync(kernel, new());
```

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs [hunk 5]
```diff
@@ -3,130 +3,109 @@
...
 
-        var textValidationFunction = kernel.CreateFunctionFromPrompt(FunctionDefinition);
+        const string Input = "Why AI is awesome";
+        Console.WriteLine($"Function input: {Input}\n");
+        var result = await ParagraphWritingFunction.InvokeAsync(kernel, new());
 
-        var result = await textValidationFunction.InvokeAsync(kernel, new("I mised the training session this morning"));
         Console.WriteLine(result.GetValue<string>());
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:54]: Small improvement:
```suggestion
        Console.WriteLine(result);
```

**Review by dmytrostruk — APPROVED** [review]

**Review by crickman — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 1 above)
Comment (glahaye) [reply dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:18]: Changed.

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 2 above)
Comment (glahaye) [reply dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:23]: Put back in

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 3 above)
Comment (glahaye) [reply dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:52]: Indeed! Thanks for catching this,

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 4 above)
Comment (glahaye) [reply dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:48]: Change taken

File: dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs (hunk 5 above)
Comment (glahaye) [reply dotnet/samples/KernelSyntaxExamples/Example16_CustomLLM.cs:54]: Thanks

**Review by gitri-ms — APPROVED** [review]

**Pull request merged**